lib/backtrace_analyzer/analyzer.rb

```diff
@@ -0,0 +1,50 @@
+module BacktraceAnalyzer
+  class Analyzer < ::BaseAnalyzer
+    def check_build(build)
+      unless build.pull_request?
+        logger.warn "Build #{build.number} is not a pull request. Skipping."
+        return
+      end
+
+      @travis.clear_session
+
+      logger.info "Analyzing PR: #{build.pull_request_number}"
+      failures = analyze_build(build)
+      report_results(build, failures)
+    end
+
+    private
+
+    def analyze_build(build)
+      failed_jobs(build).map do |job|
+        extract_backtraces(job.log)
+      end.compact
+    end
+
+    def extract_backtraces(log)
+      body = log.session.get_raw("/logs/#{log.id}")
+      failures = body.scan(/FAILED_TEST:START(.*?)FAILED_TEST:END/m)
+      errors = body.scan(/ERROR_TEST:START(.*?)ERROR_TEST:END/m)
+      (failures + errors).flatten
```

Comment:
Should we set a limit on the number of failures/errors we report directly as a comment? I think this makes a lot of sense and will save a lot of developer time when the failure/error count is low. But sometimes we'll break things in big ways and cause hundreds of failures, and in those cases it will probably take multiple commit/push cycles to fix and the comments are going to get unwieldy. How about limiting the total number of failures/errors to 10?

Reply:
I would tend to favour having a collapsed section (and I think it's <details>, not <summary>?). But we can still limit the failures displayed to something like 50 I guess? If you can come up with a better number I'll take it.